src/Symfony/Component/HttpKernel/EventListener/RouterListener.php

```diff
@@ -135,4 +151,114 @@ public static function getSubscribedEvents()
             KernelEvents::FINISH_REQUEST => array(array('onKernelFinishRequest', 0)),
         );
     }
+
+    private function getWelcomeContent()
+    {
+        $charset = ini_get('default_charset') ?: 'UTF-8';
+
+        return <<<EOF
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="$charset" />
+        <title>Welcome!</title>
+        {$this->getStylesheet()}
+    </head>
+    <body>
+        {$this->getBody()}
+    </body>
+</html>
+EOF;
+    }
+
+    private function getStylesheet()
```

Comment:
What about moving that css+body to a separate file? Right this will bloat opcache with dead code, even in prod, isn't it?

Reply:
Done.